src/core/ucc_team.c

```diff
@@ -229,3 +237,114 @@ ucc_status_t ucc_team_destroy(ucc_team_h team)
     }
     return status;
 }
+
+ucc_status_t ucc_ee_create(ucc_team_h team, const ucc_ee_params_t *params,
+                           ucc_ee_h *ee_p)
+{
+    ucc_ee_t *ee;
+
+    ee = ucc_malloc(sizeof(ucc_ee_t), "ucc execution engine");
+    if (!ee) {
+        ucc_error("failed to allocate %zd bytes for ucc execution engine",
+                  sizeof(ucc_ee_t));
+        return UCC_ERR_NO_MEMORY;
+    }
+
+    ee->team = team;
+    ee->ee_type = params->ee_type;
+    ee->ee_context_size = params->ee_context_size;
+    ee->ee_context = params->ee_context;
+    ucc_spinlock_init(&ee->lock, 0);
+    ucc_queue_head_init(&ee->event_in_queue);
+    ucc_queue_head_init(&ee->event_out_queue);
+    *ee_p = ee;
+
+    ucc_info("ucc_ee_create: ee is created ");
+
+    return UCC_OK;
+}
+
+ucc_status_t ucc_ee_destroy(ucc_ee_h ee)
+{
+    ucc_spinlock_destroy(&ee->lock);
+    ucc_free(ee);
+
+    ucc_info("ucc_ee_create: ee is destroyed ");
+
+    return UCC_OK;
+}
```

Comment:
You might have to also check and destroy during team destroy.

Reply:
you mean if user creates it but not destroyed it?

Reply:
Either library or user has to do the book keeping of allocated ee's for a given team. 

We can address this two ways: 
1) Place responsibility on the user to destroy the ee before the team destroy.
2)Library destroying the ee connected to a team, before team destroy operation.

Reply:
But then: do we also need to do the same from collective request finalize? It is also user responsibility to call finalize. At the same time we might do a bookeeping of requests allocated/started per team. We can at least keep counting (initialized, posted, finalize) and upon team destruction through a warn if initialized != finalized.